File: dotnet/azure/Mcma.Azure.BlobStorage/BlobStorageFileLocator.cs
namespace Mcma.Azure.BlobStorage
{
    public class BlobStorageFileLocator : BlobStorageLocator
    {
        public string FilePath { get; set; }

        public override string Url => $"http://{StorageAccountName}.blob.core.windows.net/{Container}/{FilePath}";
    }
}

File: azure/blob-storage/lib/blob-storage-file-locator.ts
import { BlobStorageLocator, BlobStorageLocatorProperties } from "./blob-storage-locator";

export interface BlobStorageFileLocatorProperties extends BlobStorageLocatorProperties {
    filePath: string;
}

export class BlobStorageFileLocator extends BlobStorageLocator implements BlobStorageFileLocatorProperties {
    filePath: string;

    constructor(properties: BlobStorageFileLocatorProperties) {
        super("BlobStorageFileLocator", properties);
        this.checkProperty("filePath", "string");
    }

    get url(): string {
        return `https://${this.storageAccountName}.blob.core.windows.net/${this.container}/${this.filePath}`;
    }
}
